Validation Enhancements

Rangababu-R opened the pull request:
1. Got rid of regex entirely.
2. Added item validation of the list
3. addressed validation gaps.
#42 #44 #43 #47

alakendu commented on openapiart/common.py:
```diff
@@ -187,101 +187,118 @@ def _decode(self, dict_object):
+    
+    def types_validation(self, value, type_, err_msg, itemtype=None):
+        type_map = {
+            int: "integer", str: "string",
```
list issue still present (https://github.com/open-traffic-generator/openapiart/issues/43)
I am getting error at the time of run this 
```python
    d = cfg.devices.device(name='d', container_name=p1.name)[-1]
    eth = d.ethernet
    eth.name = 'e'
    eth.mac = ['00:01:00:00:00:01']
```
```python
>           raise TypeError(err_msg)
E           TypeError: Invalid ['00:01:00:00:00:01'] format, expected mac at <class 'snappi.snappi.DeviceEthernet'>
```

Rangababu-R replied:
I think you cannot pass list to eth.mac right?

alakendu commented on openapiart/common.py:
```diff
@@ -187,101 +187,118 @@ def _decode(self, dict_object):
-            return False
-        if self._IPV6_REP1.match(ip) or self._IPV6_REP2.match(ip):
+    def validate_ipv6(self, ip):
+        if ip is None or not isinstance(ip, str):
```
Not showing any error for trailing whitespaces
```python
ip.gateway = '2a00:1450:f013:c03:0:0:0:1 '
```

Rangababu-R replied:
I feel trailing spaces can be striped, we may need to test with illegal char or space with in the address. will check that part

ajbalogh commented:
Please check the format validation cases in test_formats.py that are named test_formats_ipv4_to_be_removed and test_formats_integer_to_be_removed that demonstrate values that should not be passing.

Rangababu-R commented:
> 
> 
> Please check the format validation cases in test_formats.py that are named test_formats_ipv4_to_be_removed and test_formats_integer_to_be_removed that demonstrate values that should not be passing.

yes @ajbalogh I had added the fix for it and modified the testcase. its in my local. seeing an issue in snappi with choice. will fix and update the branch

alakendu commented on openapiart/tests/server.py:
```diff
@@ -0,0 +1,73 @@
+from flask import Flask, request, Response
```
Include "flask" in pipeline. Otherwise build will fail
